ArduCopter/mode.cpp

```diff
@@ -668,6 +668,10 @@ void Mode::land_run_horizontal_control()
                     AP::logger().Write_Event(LogEvent::LAND_REPO_ACTIVE);
                 }
                 copter.ap.land_repo_active = true;
+            } else {
+                if (copter.ap.land_repo_active && copter.precland.allow_precland_after_reposition()) {
```

Comment:
Probably need a guard to check that precland is available otherwise the copter.precland object won't exist